pixeltable/store.py

```diff
@@ -161,11 +163,56 @@ def create(self) -> None:
             else:
                 raise
 
+    def create(self) -> None:
+        """
+        Create or update store table to bring it in sync with self.sa_tbl.
+
+        If this is run outside of a transaction, runs each DDL statement in its own transaction.
+        """
+        postgres_dialect = sql.dialects.postgresql.dialect()
+        stmt = sql.schema.CreateTable(self.sa_tbl, if_not_exists=True).compile(dialect=postgres_dialect)
+        create_stmt = str(stmt)
+        stmts = [create_stmt]
+
+        for col in self.sa_tbl.columns:
+            stmts.append(self._add_column_stmt(col))
+
+        for index in self.sa_tbl.indexes:
+            stmt = sql.schema.CreateIndex(index, if_not_exists=True).compile(dialect=postgres_dialect)
+            create_stmt = str(stmt)
+            stmts.append(create_stmt)
+
+        env = Env.get()
+        for stmt in stmts:
+            if env.in_xact:
+                env.conn.execute(sql.text(stmt))
+            else:
+                # each of these statements need to be run as separate transactions; if one fails, we still want to run
+                # the rest
+                with env.begin_xact(for_write=True):
+                    self._exec_if_not_exists(stmt)
+
+    def create_index(self, idx_id: int) -> None:
+        """Create If Not Exists for this index"""
+        idx_info = self.tbl_version.get().idxs[idx_id]
```

Comment:
Keep track of `all_idxs` separately in TableVersion (a record of all indexes, including dropped indexes), and use that here to get a fully accurate representation of the store schema. We might as well, and it will solve the problem below (see additional later comment).